internal/controller/nnf_access_controller.go

```diff
@@ -1199,34 +1199,41 @@ func (r *NnfAccessReconciler) getClientMountStatus(ctx context.Context, access *
 	}
 
 	// Check whether the clientmounts have finished mounting/unmounting
+	// Track how many ClientMounts we've validated (excluding offline/fenced nodes)
+	validatedMounts := 0
 	for _, clientMount := range clientMounts {
+		// Check if this compute node is offline/fenced first, before checking mount status
+		offline, err := r.checkOfflineCompute(ctx, access, &clientMount)
+		if err != nil {
+			return false, err
+		}
+
+		log.Info("checking ClientMount status", "node", clientMount.GetNamespace(), "offline", offline, "desiredState", access.Spec.DesiredState, "statusMounts", len(clientMount.Status.Mounts), "specMounts", len(clientMount.Spec.Mounts))
+
+		// If the compute node is offline/fenced and we're unmounting, ignore any mount status
+		// from this node since it can't respond. The filesystem won't be remounted if the
+		// compute comes back since spec.desiredState is "unmounted"
+		if offline && access.Spec.DesiredState == "unmounted" {
+			log.Info("ignoring ClientMount from offline/fenced compute node during unmount", "node name", clientMount.GetNamespace())
+			validatedMounts++ // Count as validated so we don't wait for it
+			continue
+		}
+
 		if len(clientMount.Status.Mounts) != len(clientMount.Spec.Mounts) {
 			return false, nil
 		}
 
 		for _, mount := range clientMount.Status.Mounts {
 			if string(mount.State) != access.Status.State || !mount.Ready {
-				if string(mount.State) == "unmounted" {
-					offline, err := r.checkOfflineCompute(ctx, access, &clientMount)
-					if err != nil {
-						return false, err
-					}
-					// If the compute node is down, then ignore an unmount failure. If the compute node
-					// comes back up, the file system won't be mounted again since spec.desiredState is "unmounted"
-					if offline {
-						log.Info("ignoring status from offline compute node", "node name", clientMount.GetNamespace())
-						continue
-					}
-				}
-
 				return false, nil
 			}
 		}
+		validatedMounts++
 	}
 
-	if len(clientMounts) != len(clientList) {
+	if validatedMounts != len(clientList) {
```

Comment:
Is this ever different from `len(clientMounts)`? In the loop the only time we `continue` there's also a `validatedMounts++`